sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ConstantPropagationSuite.scala

```diff
@@ -37,10 +38,12 @@ class ConstantPropagationSuite extends PlanTest {
         Batch("ConstantPropagation", FixedPoint(10),
           ConstantPropagation,
           ConstantFolding,
-          BooleanSimplification) :: Nil
+          BooleanSimplification,
+          SimplifyBinaryComparison,
```

Comment:
why do we need to add these 2?

Reply:
For example when we have this test: `'a < 2 && Literal(2) === 'a` and we expect `false` as filter and so an empty relation result `BooleanSimplification` has as important role:
```
=== Applying Rule org.apache.spark.sql.catalyst.optimizer.ConstantPropagation ===
!Filter ((a#0 < 2) AND (2 = a#0))        Filter ((2 < 2) AND (2 = a#0))
 +- LocalRelation [a#0, b#1, c#2, x#3]   +- LocalRelation [a#0, b#1, c#2, x#3]
           
19/05/24 21:40:59 INFO ConstantPropagationSuite$Optimize: 
=== Applying Rule org.apache.spark.sql.catalyst.optimizer.ConstantFolding ===
!Filter ((2 < 2) AND (2 = a#0))          Filter (false AND (2 = a#0))
 +- LocalRelation [a#0, b#1, c#2, x#3]   +- LocalRelation [a#0, b#1, c#2, x#3]
           
19/05/24 21:40:59 INFO ConstantPropagationSuite$Optimize: 
=== Applying Rule org.apache.spark.sql.catalyst.optimizer.BooleanSimplification ===
!Filter (false AND (2 = a#0))            Filter false
 +- LocalRelation [a#0, b#1, c#2, x#3]   +- LocalRelation [a#0, b#1, c#2, x#3]
```

I removed `SimplifyBinaryComparison` as it is not needed, indeed.